packages/babel-plugin-transform-nullish-coalescing-operator/src/index.ts

```diff
@@ -24,10 +25,27 @@ export default declare((api, { loose = false }: Options) => {
           return;
         }
 
+        let pureMember = false;
+        if (pureGetters) {
+          let obj: t.Expression = node.left;
+          while (t.isMemberExpression(obj)) {
+            if (obj.computed) break;
+            obj = obj.object;
+          }
+          if (t.isIdentifier(obj)) {
+            pureMember = true;
+          }
```

Comment:
I would also put a limit of "at most `identifier.identifier`", since for cases like `some.object.with.a?.property` it's better to introduce the temp var than to emit `some.object.with.a == null ? null : some.object.with.a.property`